crates/nu-command/src/path/join.rs

```diff
@@ -221,14 +222,24 @@ fn join_list(parts: &[Value], head: Span, span: Span, args: &Arguments) -> Value
 
                     Value::list(vals, span)
                 }
-                Err(_) => Value::error(
-                    ShellError::PipelineMismatch {
-                        exp_input_type: "string or record".into(),
-                        dst_span: head,
-                        src_span: span,
-                    },
-                    span,
-                ),
+                Err(_) => {
+                    let wrong_type = format!(
```

Comment:
what input should I provide to test this? I couldn't find

Reply:
The error from `Value::as_record` will be a `ShellError::CantConvert` which has the original type so you can pulll that out.

Reply:
Are you saying I should replace the next ``ShellError::OnlySupportsThisInputType`` by a ``ShellError::CantConvert`` ? 
What should I use for the arguments?

Reply:
You could match on the `Err(ShellError::CantConvert` and pull out the from type and then turn it into a `Only....` for the most descriptive error message (the other `SE` variants shouldn't be raised so you could raise an `InternalError` if they come up, or in theory even panic). `CantConvert` it self isn't too bad as it is at least containing the types (but lacks the second span if I recall)

Reply:
Ah the version in `ShellError` is named `NushellFailed(Spanned)`

But I am also fine with merging your span improvements as is and dropping the remaining `PipelineMismatch` in another PR.

Reply:
Ah okay
I did that
```nushell
                Err(_) => Value::error(
                    ShellError::NushellFailed {
                        msg: "failed to join path".into(),
                    },
                    span,
                ),
```